packages/web/src/apollo/index.tsx

```diff
@@ -41,8 +96,15 @@ const ApolloProviderWithFetchConfig: React.FunctionComponent<{
    */
   apolloSetLogVerbosity(logLevel)
 
-  const { uri, headers } = useFetchConfig()
-  const { getToken, type: authProviderType, isAuthenticated } = useAuth()
+  /**
+   * Here we're using Apollo Link to customize Apollo Client's data flow.
+   *
+   * Although we're sending conventional HTTP-based requests and could just pass `uri` instead of `link`,
+   * we need to fetch a new token on every request, making middleware a good fit for this.
+   *
+   * @see {@link https://www.apollographql.com/docs/react/api/link/introduction/}
+   */
+  const { isAuthenticated, getToken, type: authProviderType } = useAuth()
 
   const withToken = setContext(async () => {
     if (isAuthenticated && getToken) {
```

Comment:
is `getToken` called twice? 

```
if (isAuthenticated && getToken) {
```

and then 

```
const token = await getToken()
```

?

Reply:
Or, the first is just seeing if the get token function exists?

Reply:
Yeah the latter (just seeing if it exists—I believe all our auth providers implement this method though). That part has been there for a while and I wasn't trying to change how the middleware actually worked at all (just trying to make it configurable) so I left it be but if you have a suggestion happy to change it!